dataherald/sql_database/models/types.py

```diff
@@ -68,11 +69,17 @@ def set_uri_without_ssh(cls, v, values):
             raise ValueError("When use_ssh is False set uri")
         return v
 
-    @validator("uri", pre=True, always=True)
+    @validator("uri", "llm_api_key", pre=True, always=True)
     def encrypt(cls, value: str):
         fernet_encrypt = FernetEncrypt()
         try:
             fernet_encrypt.decrypt(value)
             return value
         except Exception:
             return fernet_encrypt.encrypt(value)
+
+    def decrypt_api_key(self):
```

Comment:
if you use the next code every time you 
`
    @property
    def llm_api_key(self):
        if self.llm_api_key is not None and self.llm_api_key != "":
            fernet_encrypt = FernetEncrypt()
            return fernet_encrypt.decrypt(self.llm_api_key)
        return os.environ.get("OPENAI_API_KEY")    
`